### PR TITLE
Limit user access when creating new App in iTunes

### PR DESCRIPTION
**Pull request by javibm**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

I have the need to limit the user access when creating a new Application in iTunes.

I have created an app using spaceship with and without limit user access. I also used Produce to test both scenarios.

### Description

I have created a new option in Produce to pass an array of emails for limiting the user access. If any email is registered as user in iTunes, the new app is limited for these ones.

### CODE REVIEW

File: produce/lib/produce/options.rb [hunk 1]
```diff
@@ -65,6 +65,13 @@ def self.available_options
                                      description: "Skip the creation of the app on iTunes Connect",
                                      is_string: false,
                                      default_value: false),
+        FastlaneCore::ConfigItem.new(key: :itc_users,
+                                     short_option: "-s",
+                                     env_name: "ITC_USERS",
+                                     optional: true,
+                                     type: Array,
+                                     description: "Array with Itunes Connect Users for Limit User Access",
```
Comment (KrauseFx) [inline produce/lib/produce/options.rb]: This text isn't clear directly, can we rephrase to something like `Array of iTunes Connect users that have access to newly created app? If provided, no other users will have access` (at least that's how I understood it, is that correct?)

**Comment by KrauseFx** [comment]
Nice change, thanks for the contribution, I added one comment. Also, can you make sure you write `iTunes` everywhere?

**Comment by javibm** [comment]
I have updated the description to be more accurate. Also, I have checked that i write in all the create application for `iTunes`.

**Review by KrauseFx — APPROVED** [review]
Nice, thanks for updating the PR, this is a great change 👍

**Pull request merged**